src/armonik_cli/core/filter_grammar.jinja

```diff
@@ -0,0 +1,54 @@
+?start: expr
+
+expr: term ("or" term)*
+
+term: factor ("and" factor)*
+
+factor: "(" expr ")"
+    {% for field, field_comp, field_type in fields %}
+        | "{{ field }}" {{ field_comp }} {{ field_type }} -> {{ field.replace(".", "_") }}
+    {% endfor %}
+    {% if generic_option %}
+        | "options.op." STRING string_comp STRING -> generic_option
+    {% endif %}
```

Comment:
I would not use code generation for this. Also, I would handle those cases in the AST analysis instead of the grammar.